File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int i = 0;
        int j = 1;
        int maxP = 0;
        while(j < prices.size()){
            if(prices[i] < prices[j]){
                maxP = max(prices[j] - prices[i], maxP);
            }
            else{
                i = j;
            }
            j++;
        }
        return maxP;
    }
};

File: Data Structures & Algorithms/trapping-rain-water/submission-2.py
class Solution:
    def trap(self, height: List[int]) -> int:
        i = 0
        j = len(height) - 1
        leftMax = height[i]
        rightMax = height[j]
        area = 0
        while i < j:
            if leftMax < rightMax:
                i += 1
                leftMax = max(leftMax, height[i])
                area += leftMax - height[i]
            else:
                j -= 1
                rightMax = max(rightMax, height[j])
                area += rightMax - height[j]
        return area